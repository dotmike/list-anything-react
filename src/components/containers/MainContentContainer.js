import React from "react";
import PropTypes from "prop-types";

import TopNavigation from "../navigation/TopNavigation";
import MainContainer from "./MainContainer";

class MainContentContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      menuVisible: false,
      activeItem: props.location.pathname
    };
  }

  toggleMenu = (e, { to }) =>
    this.setState(prevState => ({
      menuVisible: !prevState.menuVisible,
      activeItem: to || prevState.activeItem
    }));

  hideMenu = (e, { to }) =>
    this.setState(prevState => ({
      menuVisible: false,
      activeItem: to || prevState.activeItem
    }));

  render() {
    const { children, showContent } = this.props;
    const { menuVisible, activeItem } = this.state;
    return (
      <div
        id="main-content-container"
        style={{ display: showContent ? "block" : "none" }}
      >
        {showContent && (
          <TopNavigation
            activeItem={activeItem}
            toggleMenu={this.toggleMenu}
            hideMenu={this.hideMenu}
          />
        )}
        <MainContainer
          menuVisible={menuVisible}
          activeItem={activeItem}
          hideMenu={this.hideMenu}
        >
          {children}
        </MainContainer>
      </div>
    );
  }
}

MainContentContainer.propTypes = {
  // ownProps
  showContent: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  }).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default MainContentContainer;
