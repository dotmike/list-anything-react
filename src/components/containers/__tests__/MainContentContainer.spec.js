import React from "react";
import { shallow } from "enzyme";
import MainContentContainer from "../MainContentContainer";

const defaultProps = {
  showContent: false,
  history: {
    push: () => {},
    location: {
      pathname: ""
    }
  },
  children: <span>Testando...</span>
};

describe("MainContentContainer", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<MainContentContainer {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
