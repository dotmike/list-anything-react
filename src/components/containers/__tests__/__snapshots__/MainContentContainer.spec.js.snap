// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContentContainer renders correctly 1`] = `
<div
  id="main-content-container"
  style={
    Object {
      "display": "none",
    }
  }
>
  <Connect(MainContainer)
    activeItem=""
    hideMenu={[Function]}
    menuVisible={false}
    search={[Function]}
  >
    <span>
      Testando...
    </span>
  </Connect(MainContainer)>
</div>
`;
